lib/collection/src/shard/shard_config.rs

```diff
@@ -0,0 +1,66 @@
+use crate::{CollectionError, CollectionResult, PeerId};
+use std::path::{Path, PathBuf};
+
+use segment::common::file_operations::{atomic_save_json, read_json};
+use serde::{Deserialize, Serialize};
+use std::fs;
```

Comment:
The `ShardConfig` uses sync. file operation as it is built on top of `segment::common::file_operations::{atomic_save_json, read_json}` which are sync. themselves.

Reply:
yeah, there is a limitation - https://github.com/untitaker/rust-atomicwrites/issues/38